java/src/jmri/jmrix/can/cbus/node/CbusNodeConstants.java

```diff
@@ -41,23 +44,31 @@
      * or provide extra info. which is missing for a known module firmware.
      * @param node The CbusNode object we are setting the traits for
      */
-    public static void setTraits( CbusNode node ){
+    public static void setTraits(@Nonnull CbusNode node ){
         
         // defaults
         node.setsendsWRACKonNVSET(true);
         
         if ( node.getNodeParamManager().getParameter(1) == 165 ) { // MERG MODULE
-            if ( node.getNodeParamManager().getParameter(3) == 29 ) { // CANPAN
-                node.setsendsWRACKonNVSET(false);
-            }
-            if ( node.getNodeParamManager().getParameter(3) == 10 // CANCMD
-                || node.getNodeParamManager().getParameter(3) ==  55 // or CANCSB 
-                || node.getNodeParamManager().getParameter(3) == 12 // or CANBC
-            ) { 
-                if ( node.getNodeParamManager().getParameter(7) == 4 ) { // v4 Firmware
-                    node.getNodeEventManager().resetNodeEventsToZero(); // sets num events to 0 as does not respond to RQEVN
-                    node.setStatResponseFlagsAccurate(false);
-                }
+            switch (node.getNodeParamManager().getParameter(3)) { // Module Type ID Number
+                case 29: // CANPAN
```

Comment:
No need to fix now, but it's good to have these kind of numbers as constants:

`public static final int CANPAN = 29;`

`public static final int ModuleTypeIdNumber = 3;`